statshost: OIDC as default, disable local login

dpausp opened the pull request:
@flyingcircusio/release-managers

## Release process

- [x] Created changelog entry using `./changelog.sh`

## PR release workflow (internal)

- [x] PR has internal ticket
- [x] internal issue ID (PL-…) part of branch name
- [x] internal issue ID mentioned in PR description text
- [x] ticket is on Platform agile board
- [x] ticket state set to *Pull request ready*
- [x] if ticket is more urgent than within the next few days, directly contact a member of the Platform team
<!---->
- [x] set urgency and risk labels
- [ ] ensure the merge bot has determined a merge date
- [x] ensure all checks are green
- [ ] get a review from a colleague

## Design notes

- [x] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [x] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - reduce local login accounts and use OIDC instead
  - when OIDC enabled, local login must not be enabled. Grafana redirects to Keycloak automatically
  - only applies to statshost-master, global statshost has locally managed SSO config
- [x] Security requirements tested? (EVIDENCE)
  - automated tests check that statshost-master and statshost still work
  - checked on a test VM that switching from LDAP to OIDC works, also the other way round and back to OIDC. User accounts are updated properly and show the currently enabled login method after re-login (logged-in users are not logged out automatically, that's ok)

dpausp committed:
statshost: improvements for OIDC support, statshost-master only

- disable local login form when OIDC is active
- disable basic auth system, we always want OIDC (or legacy LDAP)
- trim requested OIDC scopes, we don't need offline access and roles
  (static user mapping to Admin role)
- refactor login config to enable it only on statshost-master.
  Global statshosts have individual OIDC config for now.

PL-134048

dpausp committed:
statshost: enable OIDC by default

LDAP is now disabled by default, OIDC via our Keycloak instance is the
standard login method.

PL-134048

leona-ya commented:
I'd say that this change could lead to confusion or other behavior changes for our customers. I suggest moving this to 25.11 and introduce with this release. Now, there is no security reason to fast-track this change.

PhilTaken commented on nixos/roles/statshost/default.nix:
```diff
@@ -692,6 +635,74 @@ in
+            login_attribute_path = "preferred_username";
+            name_attribute_path = "full_name";
+            # Everybody becomes admin, ignoring user roles from OIDC.
+            role_attribute_path = "\"'Admin'\"";
```
I don't think we should assign admin to everybody considering that access to the grafana webinterface is not limited to fcio staff.

dpausp replied:
It's the same behaviour as LDAP always had, we can improve that later :)

dpausp committed:
statshost: fix OIDC role mapping

Like for LDAP, we want to assign Admin privileges to all users
logged in via OIDC. We have to escape the 'Admin' properly so it's
interpreted as a static role name instead of triggering an attribute lookup.

PL-134142
